catalog/spatial/wfs/2.0.0/spatial-wfs-v2_0_0-source/src/main/java/org/codice/ddf/spatial/ogc/wfs/v2_0_0/catalog/source/WfsSource.java

```diff
@@ -667,36 +667,32 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
                         featureCollection.getNumberReturned(), numResults);
             }
 
-            if (numResults > -1) {
-                availabilityTask.updateLastAvailableTimestamp(System.currentTimeMillis());
-                LOGGER.debug("WFS Source {}: Received featureCollection with {} metacards.",
-                        getId(), numResults);
-
-                List<Result> results = new ArrayList<Result>(numResults);
-
-                for (int i = 0; i < numResults; i++) {
-                    Metacard mc = featureCollection.getMembers().get(i);
-                    mc = transform(mc, DEFAULT_WFS_TRANSFORMER_ID);
-                    Result result = new ResultImpl(mc);
-                    results.add(result);
-                    debugResult(result);
-                }
-
-                //Fetch total results available
-                Long totalResults = new Long(0);
-                if (featureCollection.getNumberMatched() == null) {
-                    totalResults = Long.valueOf(numResults);
-                } else if (featureCollection.getNumberMatched().equals(UNKNOWN)) {
-                    totalResults = Long.valueOf(numResults);
-                } else if (StringUtils.isNumeric(featureCollection.getNumberMatched())) {
-                    totalResults = Long.parseLong(featureCollection.getNumberMatched());
-                }
+            availabilityTask.updateLastAvailableTimestamp(System.currentTimeMillis());
+            LOGGER.debug("WFS Source {}: Received featureCollection with {} metacards.",
+                    getId(), numResults);
+
+            List<Result> results = new ArrayList<Result>(numResults);
+
+            for (int i = 0; i < numResults; i++) {
+                Metacard mc = featureCollection.getMembers().get(i);
+                mc = transform(mc, DEFAULT_WFS_TRANSFORMER_ID);
+                Result result = new ResultImpl(mc);
+                results.add(result);
+                debugResult(result);
+            }
 
-                simpleResponse = new SourceResponseImpl(request, results, totalResults);
-            } else {
-                throw new UnsupportedQueryException(
-                        "The number of features returned is a negative number");
+            //Fetch total results available
+            Long totalResults = Long.valueOf(0);
```

Comment:
`Long totalResults = 0` would work as well. In fact, keeping it as a primitive `long` should be sufficient, I'd think.